assignments/sudoku/code.js

```diff
@@ -0,0 +1,25 @@
+const edgeSize = (width - height) / 2
+const b = Array(9).fill().map(() => Array(9).fill(''))
+let selected = 0;
+
+const drawBoard1st = () => {
+  for (let i = 0; i < 10; i++) {
+    const thick = i % 3 === 0 ? 3 : 1
+    drawLine(edgeSize, height / 9 * i, width - edgeSize, height / 9 * i, 'black', thick)
+    drawLine(edgeSize + height / 9 * i, 0, edgeSize + height / 9 * i, height, 'black', thick)
+    drawText(i, edgeSize / 10 * i, 20, 'black', 20)
+  }
+}
+
+registerOnclick((x, y) => {
+  if (x < edgeSize && y < 20) {
+    selected = Math.floor(x / (edgeSize / 10))
+  } else if (x > edgeSize && x < width - edgeSize) {
+    const exe = Math.floor((x - edgeSize) / (height / 9))
+    const wiy = Math.floor(y / (height / 9))
```

Comment:
I know you love these names but you could also use simple names that also show the transition from graphical coordinates (x, y) to sudoku coordinates of row and column. 

```suggestion
    const row = Math.floor((x - edgeSize) / (height / 9))
    const col = Math.floor(y / (height / 9))
```

This also has the nice effect that when you index into the board it's in the normal order we talk about rows and columns: `b[row][col]` as opposed to backwards `b[wiy][exe]` when we normally talk about x,y coordinates, not y,x coordinates. Finally `exe` is commonly used in computing as an abbreviation for "executable" so most programmers will probably not read it as `x` and will wonder what you're on about.